functionary/ui/templates/forms/environment_selector.html

```diff
@@ -1,25 +1,55 @@
-                    </option>
-                {% endfor %}
-            </optgroup>
+    {% for team, envs in user_environments.items %}
+        {% for env in envs %}
+            {% if environment_id == env.id|stringformat:'s' %}
+                <button type="button"
+                        class="btn btn-primary btn-block text-start ps-2"
+                        title="Change Environment"
+                        data-bs-toggle="dropdown"
+                        aria-expanded="false">
+                    <i class="fa fa-lg fa-building fa-fw"></i>
+                    <span class="ms-1 d-none d-md-inline" id="btnLabel">{{ env.name }}</span>
+                    <span class="visually-hidden">Toggle Dropdown</span>
+                </button>
+                <ul class="dropdown-menu shadow-lg dropdown-menu-dark">
+                    {% for team, envs in user_environments.items %}
+                        <li>
+                            <span class="dropdown-header" type="button">{{ team }}</span>
+                        </li>
+                        {% for env in envs %}
+                            <li>
+                                <button class="dropdown-item {% if environment_id == env.id|stringformat:'s' %}fw-semibold active{% endif %}"
+                                        type="button"
+                                        title="{{ env.name }}"
+                                        {% if environment_id == env.id|stringformat:'s' %}{% endif %}
+                                        hx-post="{% url 'ui:set-environment' %}"
+                                        hx-vals='{"environment_id": "{{ env.id }}"}'
+                                        hx-swap="none">
+                                    {{ env.name }}
+                                </button>
+                            </li>
+                        {% endfor %}
+                    {% endfor %}
+                </ul>
+                <button type="button"
+                        title="Management"
+                        class="btn btn-primary btn-block dropdown-toggle dropdown-toggle-split"
+                        data-bs-toggle="dropdown"
+                        aria-expanded="false">
+                    <i class="fa fa-ellipsis"></i>
+                    <span class="visually-hidden">Toggle Dropdown</span>
+                </button>
+                <div class="dropdown-menu shadow-lg dropdown-menu-dark">
+                    <a href="{% url 'ui:environment-detail' pk=environment_id %}"
+                       class="dropdown-item"
+                       title="Manage Environment">
+                        <i class="fa fa-lg fa-building fa-fw me-1"></i>Manage Environment</a>
+                    <a href="{% url 'ui:team-detail' pk=env.team.id %}"
+                       class="dropdown-item"
+                       title="Manage Team">
+                        <i class="fa fa-lg fa-users fa-fw me-1"></i>Manage Team</a>
```

Comment:
Minor nit, but the `fa-lg` feels too big here.  Playing around with it I kind of think going all the way down to `fa-sm` looks best.